src/caretogether-pwa/src/Volunteers/VolunteerApproval/CustomFieldsFilter.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  CombinedFamilyInfo,
+  CustomField,
+  CustomFieldType,
+} from '../../GeneratedClient';
+import { CustomFieldsFilterSelect } from './CustomFieldsFilterSelect';
+
+type VolunteerFilterProps = {
+  customFamilyFields: CustomField[];
+  volunteerFamilies: CombinedFamilyInfo[];
+  customFieldFilters: Record<string, string[]>;
+  changeCustomFieldFilter: (fieldName: string, value: string[]) => void;
+};
+
+export function CustomFieldsFilter({
+  customFamilyFields,
+  volunteerFamilies,
+  customFieldFilters,
+  changeCustomFieldFilter,
+}: VolunteerFilterProps) {
+  return customFamilyFields?.map(({ name: fieldName, type: fieldType }) => {
+    if (!fieldName) return null;
+
+    const uniqueValues = Array.from(
+      new Set(
+        volunteerFamilies
+          .map((family) => {
+            const field = family.family?.completedCustomFields?.find(
+              (customField) => customField.customFieldName === fieldName
+            );
+            return field?.value;
+          })
+          .filter((value) => value !== null && value !== undefined)
+      )
```

Comment:
The text-based value suggestions for the custom fields in the filters need to be sorted alphabetically. The one live organization I tested has 42 values for "Home Church" currently, and they all display in semi-random order. 😅